samples/highcharts/demo/music-race-chart/demo.js

```diff
@@ -0,0 +1,196 @@
+        },
+        labels: {
+            format: '${text} B'
+        }
+    },
+    tooltip: {
+        pointFormat: '{series.name} had a revenue of <b>${point.y:,.1f} B</b><br/> in {point.x}'
+    },
+    plotOptions: {
+        area: {
+            stacking: 'normal',
+            pointStart: startYear,
+            pointEnd: endYear,
+            marker: {
+                enabled: false
+            },
+
+            label: {
+                style: {
+                    transition: 'transform 0.4s, opacity 0.8s'
+                }
+            }
+        }
+    },
+
+    series: [{
+        label: {
+            style: {
+                fontSize: '0.8em'
+            }
+        }
+    }, {
+        label: {
+            style: {
+                fontSize: '0.3em'
+            }
+        }
+    }, {
+        label: {
+            style: {
+                fontSize: '0.4em'
+            }
+        }
+    }, {
+        label: {
+            style: {
+                fontSize: '2em'
+            }
+        }
+    }, {
+        label: {
+            style: {
+                fontSize: '0.5em'
+            }
+        }
+    }, {
+        label: {
+            style: {
+                fontSize: '0.5em'
+            }
+        }
+    }]
+
+});
+
+function pause(button) {
+    button.title = 'play';
+    button.className = 'fa fa-play';
+    clearTimeout(chart.sequenceTimer);
+    chart.sequenceTimer = undefined;
+}
+
+function update() {
+    chart.update({
+        subtitle: {
+            text: 'Year ' + input.value
+        }
+    },
+    false,
+    false,
+    false);
+
+    const series = chart.series,
+        yearIndex = input.value - startYear,
+        dataLength = series[0].data.length;
+
+    // If slider moved back in time
+    if (yearIndex < dataLength - 1) {
+        for (let i = 0; i < series.length; i++) {
+            const seriesData = series[i].data.slice(0, yearIndex);
+            series[i].setData(seriesData, false);
+        }
+    }
+
+    // If slider moved forward in time
+    if (yearIndex > dataLength - 1) {
+        const remainingYears = yearIndex - dataLength;
+        for (let i = 0; i < series.length; i++) {
+            for (let j = input.value - remainingYears; j < input.value; j++) {
+                series[i].addPoint([getRevenue(formatRevenue[i], j)], false);
+            }
+        }
+    }
+
+    // Add current year
+    for (let i = 0; i < series.length; i++) {
+        series[i].addPoint([getRevenue(formatRevenue[i], input.value)], false);
+    }
+
+    chart.redraw();
+
+    input.value = parseInt(input.value, 10) + 1;
+
+    if (input.value > endYear) {
+        // Auto-pause
+        pause(btn);
+    }
+
+}
+
+function play(button) {
+    // Reset slider at the end
+    if (input.value > endYear) {
+        input.value = startYear;
+    }
+    button.title = 'pause';
+    button.className = 'fa fa-pause';
+    chart.sequenceTimer = setInterval(function () {
+        update();
+    }, 700);
+}
```

Comment:
On the contrary to our other race chart demos, this one has no visible data to start with. So I think it would be better to auto-play the race, otherwise users will be looking at an empty canvas until they find the play button.